Makefile

```diff
@@ -42,22 +41,85 @@ godoc: ## runs godoc server and opens in browser
 	@# The read varname command will keep the make command running (waiting for input) until you press CTRL+C.
 	@trap 'kill `cat godoc.pid` && rm -f godoc.pid' EXIT; read varname
 
+.PHONY: fmt
 fmt: ## formats all go files
 	go fmt ./...
 	make format-lines
 
+.PHONY: format-lines
 format-lines: ## formats all go files with golines
 	go install github.com/segmentio/golines@latest
 	golines -w -m 120 --ignore-generated --shorten-comments --ignored-dirs=${GO_LINES_IGNORED_DIRS} ${GO_FOLDERS}
 
+.PHONY: lint
 lint: ## runs all linters
 	go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
 	golangci-lint run ./...
 
+
+___BINDINGS___: ## 
+
+core_default := "DelegationManager IRewardsCoordinator ISlasher StrategyManager EigenPod EigenPodManager IStrategy IAVSDirectory"
+core_location := "./lib/eigenlayer-middleware/lib/eigenlayer-contracts"
+core_bindings_location := "../../../../bindings"
+
+middleware_default := "RegistryCoordinator IndexRegistry OperatorStateRetriever StakeRegistry BLSApkRegistry IBLSSignatureChecker ServiceManagerBase IERC20"
+middleware_location := "./lib/eigenlayer-middleware"
+middleware_bindings_location := "../../bindings"
+
+sdk_default := "MockAvsServiceManager ContractsRegistry"
+sdk_location := "."
+sdk_bindings_location := "./bindings"
+
+.PHONY: core-bindings ## generates core contracts bindings
+core-bindings: ## generates core bindings
+	@echo "Starting core bindings generation"
+ifneq ($(contracts),)
+	@echo "Contracts: $(contracts)"
+	cd contracts && ./generate-bindings.sh $(core_location) $(contracts) $(core_bindings_location)
+else
+	@echo "Contracts: $(core_default)"
+	cd contracts && ./generate-bindings.sh $(core_location) $(core_default) $(core_bindings_location)
+endif
+
+
+.PHONY: middleware-bindings ## generates middleware contracts bindings
+middleware-bindings: ## generates middleware bindings
+	@echo "Starting middleware bindings generation"
+ifneq ($(contracts),)
+	@echo "Contracts: $(contracts)"
+	cd contracts && ./generate-bindings.sh $(middleware_location) $(contracts) $(middleware_bindings_location)
+else
+	@echo "Contracts: $(middleware_default)"
+	cd contracts && ./generate-bindings.sh $(middleware_location) $(middleware_default) $(middleware_bindings_location)
+endif
+
+.PHONY: sdk-bindings ## generates sdk contracts bindings
+sdk-bindings: ## generates sdk bindings
+	@echo "Starting sdk bindings generation"
+ifneq ($(contracts),)
+	@echo "Contracts: $(contracts)"
+	cd contracts && ./generate-bindings.sh $(sdk_location) $(contracts) $(sdk_bindings_location)
+else
+	@echo "Contracts: $(middleware_default)"
+	cd contracts && ./generate-bindings.sh $(sdk_location) $(sdk_default) $(sdk_bindings_location)
+endif
+
+.PHONY: bindings
+bindings: ## generates all contract bindings
+	rm -rf bindings/* && make core-bindings middleware-bindings sdk-bindings
```

Comment:
Why not also eigenpod bindings?

Reply:
because I am lazy. I will add it later - I added future improvements. Also I am not adding eigenpods since it's a separate process (different branch)

Reply:
fa01f4851fe77da845925b317787eae58f92de22